### PR TITLE
Add error on Bootstrap form

### PR DESCRIPTION
**Pull request by madeindjs**
Hello,

I loved your work so I decided to help you to close a ticket #226 

I also add documentation to create a local environnement.

Hope it help you :)

### CODE REVIEW

**Comment by potomak** [comment]
Cool! Thanks!

Can you attach a list of visual changes (screenshots) introduced by this update please?

**Comment by potomak** [comment]
If it doesn't introduce any change to the HTML can you add some view tests that prove that please?

File: app/views/shared/_form_errors.html.erb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<% if model.errors.any? %>
+<div id="error_explanation">
+  <h2><%= pluralize(model.errors.count, "error") %> prohibited this <%= model.class.name %> from being saved:</h2>
```
Comment (potomak) [inline app/views/shared/_form_errors.html.erb]: What would you think of adding a new translation for this string?

Note: please use `model.class.model_name.human`.

File: setup_local_environnement.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# Setup your local environnement
```
Comment (potomak) [inline setup_local_environnement.md]: environ~ne~ment

Comment (potomak) [reply setup_local_environnement.md]: Ignore this comment. I'd like to include just the bootstrap_form update in this PR.

**Comment by potomak** [comment]
This PR looks great!

I'm not a big bootstrap fan, but it get things done fast. Using `bootstrap_form` abstracts away from it, so I really approve this update!

I'll test myself this branch during the week. I'd like to include this update in a new release by the end of this week.

**Comment by potomak** [comment]
@madeindjs one last note. I'd like to introduce just the bootstrap_form update in this PR. Would you be so kind to remove `setup_local_environnement.md` and rebase the branch please?

File: setup_local_environnement.md (hunk 2 above)
Comment (madeindjs) [reply setup_local_environnement.md]: Sorry it's a french word... Do you prefer to include on this file on [this issue](https://github.com/tomatoes-app/tomatoes/issues/218) or simply remove it?

Comment (potomak) [reply setup_local_environnement.md]: Before working on that issue, I would like to complete https://github.com/tomatoes-app/tomatoes/issues/128 to support Docker. Just remove this file for now please.

Comment (madeindjs) [reply setup_local_environnement.md]: I removed file https://github.com/tomatoes-app/tomatoes/pull/299/commits/41c51892ee1ffa2753cc856b4afc075cf8d1e55a

File: app/views/shared/_form_errors.html.erb (hunk 1 above)
Comment (madeindjs) [reply app/views/shared/_form_errors.html.erb]: I replace `model.class.name` on https://github.com/tomatoes-app/tomatoes/pull/299/commits/c3424ced55d655cf34bbbe4825597c7e42a11286.

**Comment by potomak** [comment]
@madeindjs thanks for the updates. I'm sorry, but I didn't have the time to test the PR, I will do it as soon as possible though.

**Comment by potomak** [comment]
@madeindjs I think I will rewrite the history of this branch before merging it in though, because there are couple of commits that contradict each other, for instance 1c37c16a4143016806107aa0ee0b65e0a60f98e0 (Add instalation guide) and 41c51892ee1ffa2753cc856b4afc075cf8d1e55a (Remove setup file).

**Comment by madeindjs** [comment]
@potomak no problems ;)

File: app/views/projects/_form.html.erb [hunk 3]
```diff
@@ -1,49 +1,9 @@
-<%= form_for(@project, html: { class: 'form-horizontal' }) do |f| %>
-  <% if @project.errors.any? %>
-    <div id="error_explanation">
-      <h2><%= pluralize(@project.errors.count, "error") %> prohibited this project from being saved:</h2>
+<%= bootstrap_form_for @project, class: 'form-horizontal' do |f| %>
```
Comment (dalpo) [inline app/views/projects/_form.html.erb]: @madeindjs Could you remove the `class: 'form-horizontal'`. The `bootstrap_form_for` does not support the class option. Moreover this is currently renderering a bootstrap "vertical" style form (not an horizontal form) :smile:

File: app/views/tomatoes/_form.html.erb [hunk 4]
```diff
@@ -1,21 +1,8 @@
...
-    </div>
-  <% end %>
-
-  <div class="form-group">
-    <%= f.label :tag_list %>
-    <%= f.text_field :tag_list, class: 'form-control input-lg', placeholder: 'cook, dessert' %>
-  </div>
-  <%= f.submit class: 'btn btn-default', :'data-disable-with' => t('saving') %>
+  <%= f.text_field :tag_list, class: 'form-control input-lg', placeholder: 'cook, dessert' %>
+  <%= f.submit :'data-disable-with' => t('saving') %>
```
Comment (dalpo) [inline app/views/tomatoes/_form.html.erb]: There is a little regression on the submit button (see the image below). Could you fix it?

screenshot from 2017-09-20 09-34-03

File: app/views/projects/_form.html.erb (hunk 3 above)
Comment (potomak) [reply app/views/projects/_form.html.erb]: Great catch! I think we could use the `:horizontal` option, see https://github.com/bootstrap-ruby/rails-bootstrap-forms/blob/0fac4b176e753e2883b6632ab62926981cc88550/lib/bootstrap_form/helper.rb#L16.

File: app/views/tomatoes/_form.html.erb (hunk 4 above)
Comment (potomak) [reply app/views/tomatoes/_form.html.erb]: Another great catch! 💯  for the screenshot!

`submit` should include those classes by default, see https://github.com/bootstrap-ruby/rails-bootstrap-forms/blob/0fac4b176e753e2883b6632ab62926981cc88550/lib/bootstrap_form/helpers/bootstrap.rb#L5, do you have any idea why this doesn't render as expected @madeindjs?

**Comment by potomak** [comment]
@madeindjs do you have some time to work on the issues pointed out by @dalpo please?

**Comment by madeindjs** [comment]
@potomas, I will try to take a look on this next week ;)

File: app/views/tomatoes/_form.html.erb (hunk 4 above)
Comment (madeindjs) [reply app/views/tomatoes/_form.html.erb]: I think that Rails Bootstrap Forms does not support `data-disable-with` attribute. I will try use simple [`submit_tag`](http://api.rubyonrails.org/classes/ActionView/Helpers/FormTagHelper.html#method-i-submit_tag) method

File: app/views/projects/_form.html.erb (hunk 3 above)
Comment (madeindjs) [reply app/views/projects/_form.html.erb]: Corrected on https://github.com/madeindjs/tomatoes/commit/0cdd26ba4d930a72e8b640ec844f623ecceffe5f

File: app/views/tomatoes/_form.html.erb (hunk 4 above)
Comment (madeindjs) [reply app/views/tomatoes/_form.html.erb]: I don't see regression but I changed submit button and I used [`submit_tag`](http://api.rubyonrails.org/classes/ActionView/Helpers/FormTagHelper.html#method-i-submit_tag) method on https://github.com/madeindjs/tomatoes/commit/5ccb329aabd1e35c7d6e13a3dcb1f4f73c3cf3fe. Is this commit fix your issue?

Comment (potomak) [reply app/views/tomatoes/_form.html.erb]: Can you run the app and take a screenshot @madeindjs please?

Comment (potomak) [reply app/views/tomatoes/_form.html.erb]: > I think that Rails Bootstrap Forms does not support `data-disable-with` attribute

I don't think so, it should be just proxying options to a `submit_form` call, but I'd be happy to be wrong.

Comment (madeindjs) [reply app/views/tomatoes/_form.html.erb]: There is (in french) tag

Comment (potomak) [reply app/views/tomatoes/_form.html.erb]: @dalpo can you take one last look to this PR before approval?

**Comment by dalpo** [comment]
It seems good to me!
@madeindjs Thanks a lot fo your help :smiley:

**Comment by potomak** [comment]
@dalpo many things were missing, some other where just plain wrong. Please take one last look at this before I merge it.

cc @madeindjs @bugant

**Comment by potomak** [comment]
@dalpo I'm going to merge it without your approval...

**Pull request merged**